### PR TITLE
feat: introduce new EncodeReveal error

### PR DESCRIPTION
**Pull request by tmpolaczyk**
As a fix for #2312

### CODE REVIEW

**Comment by aesedepece** [comment]
This change sounds like a good candidate for being shipped with 1.6 — behind a TAPI flag if necessary. What do you think?

**Comment by tmpolaczyk** [comment]
Sure, this needs TAPI, and a WIP to add the new radon error.

**Comment by tmpolaczyk** [comment]
I can write this one.

File: rad/src/types/mod.rs [hunk 1]
```diff
@@ -466,13 +466,25 @@ impl TryFrom<CborValue> for RadonTypes {
...
         |(slice, inner)| match panic::catch_unwind(|| RadonTypes::try_from(slice)) {
-            Ok(Ok(radon_types)) => Some(RadonReport::from_result(
-                Ok(radon_types),
-                &ReportContext::default(),
-            )),
+            Ok(Ok(radon_types)) => {
+                // Handle future errors that should not appear yet
+                if let RadonTypes::RadonError(radon_error) = &radon_types {
+                    if let RadError::EncodeReveal = radon_error.inner() {
+                        if active_wips.wip0026() {
```
Comment (tmpolaczyk) [inline rad/src/types/mod.rs:477]: This is backwards, we want to decode a MalformedReveal before WIP activation, not after.

```suggestion
                        if !active_wips.wip0026() {
```

File: validations/src/tests/mod.rs [hunk 2]
```diff
@@ -7790,6 +7790,74 @@ fn tally_unhandled_intercept_mode_tie_has_no_message() {
...
+    let x = validate_tally_transaction(
+        &tally_transaction_error_encode_reveal,
+        &dr_pool,
+        ONE_WIT,
+        &active_wips,
+    )
+    .map(|_| ());
+    x.unwrap();
+}
+
```
Comment (aesedepece) [inline validations/src/tests/mod.rs:7860]: Thanks for this test.

File: validations/src/tests/mod.rs [hunk 3]
```diff
@@ -7790,6 +7790,74 @@ fn tally_unhandled_intercept_mode_tie_has_no_message() {
     x.unwrap();
 }
 
+#[test]
+fn tally_error_encode_reveal_wip() {
+    // Reveal value (EncodeReveal error): 39([97])
+    let reveal_value = vec![216, 39, 129, 24, 97];
+    let dr_output = example_data_request_output(2, 200, 20);
+    //let (dr_pool, dr_pointer, rewarded, slashed, error_witnesses, _dr_pkh, change, reward) =
+    //dr_pool_with_dr_in_tally_stage_only_errors(dr_output, 1, reveal_values);
```
Comment (aesedepece) [inline validations/src/tests/mod.rs:7799]: Redundant?

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: validations/src/tests/mod.rs (hunk 3 above)
Comment (tmpolaczyk) [reply validations/src/tests/mod.rs:7799]: Yes

**Review by aesedepece — APPROVED** [review]
This seems correct to me, and completely in line with the [WIP-0026](https://github.com/witnet/WIPs/blob/master/wip-0026.md) specification.

**Review by aesedepece — APPROVED** [review]

**Pull request merged**